lang/java/pom.xml

```diff
@@ -43,51 +43,49 @@
       -Dhadoop.version=1 or leave unspecified to build against Hadoop 2
     -->
     <hadoop1.version>1.2.1</hadoop1.version>
-    <hadoop2.version>2.5.1</hadoop2.version>
+    <hadoop2.version>2.7.3</hadoop2.version>
     <jackson.version>1.9.13</jackson.version>
     <jetty.version>6.1.26</jetty.version>
     <jetty-servlet-api.version>2.5-20081211</jetty-servlet-api.version>
-    <jopt-simple.version>4.7</jopt-simple.version>
-    <junit.version>4.11</junit.version>
+    <jopt-simple.version>5.0.3</jopt-simple.version>
+    <junit.version>4.12</junit.version>
     <netty.version>3.5.13.Final</netty.version>
-    <paranamer.version>2.7</paranamer.version>
-    <protobuf.version>2.5.0</protobuf.version>
-    <thrift.version>0.9.1</thrift.version>
-    <slf4j.version>1.7.7</slf4j.version>
-    <snappy.version>1.1.1.3</snappy.version>
+    <paranamer.version>2.8</paranamer.version>
+    <protobuf.version>2.6.1</protobuf.version>
```

Comment:
this will definitely be breaking, given the history of compatibility issues amongst Protobuf minor versions. Probably fine if we're targeting 1.9